core/tee/tee_fs_common.c

```diff
@@ -391,14 +391,15 @@ static int read_and_decrypt_file(int fd,
 	if (!ciphertext) {
 		EMSG("Failed to allocate file data buffer, size=%zd",
 				file_size);
-		return -1;
+		res = FD_MAP_ERR(TEE_ERROR_OUT_OF_MEMORY);
```

Comment:
Even if I proposed this when discussing with Cedric, I wonder if it is a so good way to return error. Another way would be

* return -1 in case of an error
* add a TEE_Result * argument, which is the type of error

@jforissier @jamescfkung What do you think?